pkg/controllers/state/suite_test.go

```diff
@@ -64,17 +67,55 @@ var _ = AfterSuite(func() {
 })
 
 var _ = BeforeEach(func() {
+	// fmt.Printf("Running setup...\n")
 	cloudProvider = &fake.CloudProvider{InstanceTypes: fake.InstanceTypesAssorted()}
 	cluster = state.NewCluster(env.Client, cloudProvider)
+	provisioner = test.Provisioner(test.ProvisionerOptions{ObjectMeta: metav1.ObjectMeta{Name: "default"}})
 	nodeController = state.NewNodeController(env.Client, cluster)
 	podController = state.NewPodController(env.Client, cluster)
-	provisioner = test.Provisioner(test.ProvisionerOptions{ObjectMeta: metav1.ObjectMeta{Name: "default"}})
+	metricScraper = statemetrics.NewMetricScraper(ctx, cluster)
```

Comment:
Can we test this separately, since it's decoupled?

Reply:
Since we're moving it to `pkg/controllers/metrics`, I'll move the unit tests as well.